File: 547-number-of-provinces/number-of-provinces.cpp
class Solution {
private:
    void DFS (vector<vector<int>> graph, int start, vector<bool> &check, int size) {
        check[start] = 1;
        for (int i = 0 ; i < size;i++) {
            if (graph[start][i] && !check[i]) {
                graph[i][start] = 0;
                DFS(graph, i , check, size);
            }
            
        }
    }
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
        int i = 0;
        int res = 0;
        int size = isConnected.size();
        vector<bool> boolien(size, 0);
        while (i < size) {
            isConnected[i][i] = 0;
            if (boolien[i] == 0) {
                DFS(isConnected, i , boolien, size);
                res++;
            }
            i++;
        }
        return res;
    }
};

File: 547-number-of-provinces/number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        i = 0
        res = 0
        size = len(isConnected)
        boolien = [0] * size
        while (i < size):
            isConnected[i][i] = 0
            if boolien[i] == False:
                DFS(isConnected, i , boolien, size)
                res+=1
            i+=1
        
        return res

def DFS (graph, start, boolien, size):
    boolien[start] = 1
    for i in range(0, size):
        if (graph[start][i] == 1 and boolien[i] == 0) :
            DFS(graph, i, boolien, size)